zerver/lib/event_schema.py

```diff
@@ -1550,13 +1542,26 @@ def check_update_global_notifications(
             ]
         ),
     ),
-    ("stream_id", int),
-    ("stream_name", str),
+    (ORIG_TOPIC, str),
+]
```

Comment:
This is an improvement already, but I think one could improve further with splitting `update_stream_message_fields`, i.e. the fields present whenever a stream message is modified, from `update_message_change_stream_fields`, and probably have that choice replace the `has_new_stream_id` parameter below.

Reply:
See #20611 for that improvement / clarification of the event schema.